src/mozanalysis/frequentist_stats/linear_models/functions.py

```diff
@@ -17,6 +18,35 @@
 
 logger = logging.getLogger(__name__)
 
+##Types
+
+ComparativeOption = Literal["individual", "comparative"]
+
+Estimates: TypeAlias = pd.Series
+BranchLabel = str
+EstimatesByBranch = dict[BranchLabel, Estimates]
+
+CompareBranchesOutput = dict[ComparativeOption, EstimatesByBranch]
+
+Uplift = Literal["abs_uplift", "rel_uplift"]
+
+## Exceptions
+
+
+class CovariateNotFound(Exception):
+    pass
+
+
+class UnableToAnalyze(Exception):
+    pass
+
+
+class FailedToFitModel(Exception):
+    pass
```

Comment:
You could have these inherit from a `CovariateException` or something, which would inherit from `Exception`, if you want to group them in some way. Not sure if that accomplishes anything meaningful though if they are all just `Exception` by a different name.

Reply:
Done